contracts/root/TokenPredicates/MintableERC721Predicate.sol

```diff
@@ -0,0 +1,125 @@
+pragma solidity ^0.6.6;
+
+import {IERC721Receiver} from "@openzeppelin/contracts/token/ERC721/IERC721Receiver.sol";
+import {AccessControl} from "@openzeppelin/contracts/access/AccessControl.sol";
+import {RLPReader} from "../../lib/RLPReader.sol";
+import {IMintableERC721} from "../RootToken/IMintableERC721.sol";
+import {ITokenPredicate} from "./ITokenPredicate.sol";
+import {Initializable} from "../../common/Initializable.sol";
+
+contract MintableERC721Predicate is ITokenPredicate, AccessControl, Initializable, IERC721Receiver {
+    using RLPReader for bytes;
+    using RLPReader for RLPReader.RLPItem;
+
+    // keccak256("MANAGER_ROLE")
+    bytes32 public constant MANAGER_ROLE = 0x241ecf16d79d0f8dbfb92cbc07fe17840425976cf0667f022fe9877caa831b08;
+    // keccak256("MintableERC721")
+    bytes32 public constant TOKEN_TYPE = 0xd4392723c111fcb98b073fe55873efb447bcd23cd3e49ec9ea2581930cd01ddc;
+    // keccak("Transfer(address,address,uint256)")
+    bytes32 public constant TRANSFER_EVENT_SIG = 0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef;
+
+    event LockedMintableERC721(
+        address indexed depositor,
+        address indexed depositReceiver,
+        address indexed rootToken,
+        uint256 tokenId
+    );
+
+    modifier only(bytes32 role) {
+        require(hasRole(role, _msgSender()), "MintableERC721Predicate: INSUFFICIENT_PERMISSIONS");
+        _;
+    }
+
+    constructor() public {}
+
+    function initialize(address _owner) external initializer {
+        _setupRole(DEFAULT_ADMIN_ROLE, _owner);
+        _setupRole(MANAGER_ROLE, _owner);
+    }
+
+    /**
+     * @notice accepts safe ERC721 transfer
+     */
+    function onERC721Received(
+        address,
+        address,
+        uint256,
+        bytes calldata
+    )
+        external
+        override
+        returns (bytes4)
+    {
+        return IERC721Receiver.onERC721Received.selector;
+    }
+
+    /**
+     * @notice Lock ERC721 tokens for deposit, callable only by manager
+     * @param depositor Address who wants to deposit token
+     * @param depositReceiver Address (address) who wants to receive token on child chain
+     * @param rootToken Token which gets deposited
+     * @param depositData ABI encoded tokenId
+     */
+    function lockTokens(
```

Comment:
This function just transfers the tokens to predicate from the person who sent tx. It doesn't really do anything about the for address. So this might be fine only.